sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -612,16 +612,9 @@ public static String[] getKeysFromToken(String token) {
         return keys;
     }
 
-    public static <T> T callWithOptionalTimeout(Supplier<T> callable, ExecutorService threadPool, Duration timeout,
-        ClientLogger logger) {
-        try {
-            return hasTimeout(timeout)
-                ? getResultWithTimeout(threadPool.submit(callable::get), timeout)
-                : callable.get();
-        } catch (InterruptedException | ExecutionException | TimeoutException ex) {
-            throw logger.logExceptionAsError(new RuntimeException(ex));
-        } catch (RuntimeException ex) {
-            throw logger.logExceptionAsError((RuntimeException) mapThrowableToTableServiceException(ex));
```

Comment:
We should still catch and handle an unexpected `RuntimeException` and map it to the appropriate `TableServiceException`.

Reply:
Also, maybe it would be easier to leave this and the client methods as they currently are and only update `createTableIfNotExists()` to include:
```java
try {
    return hasTimeout(timeout)
        ? getResultWithTimeout(threadPool.submit(callable::get), timeout)
        : callable.get();
} catch (RuntimeException e) {
    throw logger.logExceptionAsError((RuntimeException) mapThrowableToTableServiceException(ex));
}
```

Reply:
Another alternative can be adding a new `List<Integer> statusCodesToIgnore` parameter (or something with a better name) to this method to handle swallowing exceptions here instead of on the separate `swallow404Exception(Throwable)` method.

Reply:
I like the approach of removing the `RuntimeException` catch from the call helper and placing the responsibility on the methods themselves to allow for each method to have a uniquely defined way to handle `RuntimeException`. It's more cost effective for when adjustments are made in the future.